Content/Tools/Developer/APITool.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DragonLens.Core.Systems.ThemeSystem;
+using DragonLens.Core.Systems.ToolSystem;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using ReLogic.Content;
+using System;
+using Terraria;
+using Terraria.Localization;
+using Terraria.ModLoader;
+
+internal sealed class APITool : Tool
+{
+	private readonly Asset<Texture2D> _icon;
+	private readonly Action _click;
+	private readonly string _key;
+
+	public APITool(Mod parent, string key, Asset<Texture2D> icon, Action onClick)
+	{
+		_key = key;
+		_icon = icon;
+		_click = onClick;
+
+		// These few commands register the tool.
+		// It essentially does exactly what Tool.Register() does
+		ThemeHandler.RegisterAPIIcon(_key, _icon);
+		keybind = KeybindLoader.RegisterKeybind(parent, key, Keys.None);
+		Language.GetOrRegister($"Mods.{parent.Name}.Tools.{key}.DisplayName", () => key);
+
+		// make DragonLens aware of this tool
+		ToolHandler.AddTool(this);
+		ModTypeLookup<Tool>.Register(this);   // keeps reflection helpers happy
```

Comment:
Putting behavior like this in constructors is a bit iffy, especially since these dont enforce a singleton pattern. This seems to be an unfortunate side effect of using Mod.Call to implement an API like this, but if its an absolute must than this behavior should probably be directly in the call function and not here, to ensure its only ran once